### PR TITLE
optimize display tracks menu

### PR DESCRIPTION
**Pull request by dyphire**
ref:
image

### CODE REVIEW

File: lua/dyn_menu.lua [hunk 1]
```diff
@@ -16,9 +16,24 @@
 local menu_prop = 'user-data/menu/items'
 local menu_items = mp.get_property_native(menu_prop, {})
 
+local function esc_for_codec(str)
+    if str:find("mpeg2") then str = "mpeg2"
```
Comment (tsl0922) [inline lua/dyn_menu.lua:20]: Why not just return here?

File: lua/dyn_menu.lua [hunk 2]
```diff
@@ -16,9 +16,24 @@
...
+    elseif str:find("pcm") then str = "pcm"
+    elseif str:find("pgs") then str = "pgs"
+    elseif str:find("subrip") then str = "srt"
+    elseif str:find("vtt") then str = "vtt"
+    elseif str:find("dvd_sub") then str = "vob_sub"
+    elseif str:find("dvb_sub") then str = "dvb_sub"
+    elseif str:find("dvb_tele") then str = "teletext"
+    elseif str:find("arib") then str = "arib"
+    end
+    return str
```
Comment (tsl0922) [inline lua/dyn_menu.lua:31]: put it in `else` block?

**Comment by tsl0922** [comment]
Do you have any sources for these abbreviations, or it's just file extension?

I agree it may be more readable for `subrip` as `srt`, but not sure if others are too.

**Comment by dyphire** [comment]
> Do you have any sources for these abbreviations, or it's just file extension?
> 
> I agree it may be more readable for `subrip` as `srt`, but not sure if others are too.

It basically comes from mpv.net, ref [src/MpvNet/Player.cs#L769-L773](https://github.com/mpvnet-player/mpv.net/blob/ad94042a2c20a4daffaba0574b0c4eebc636c0d4/src/MpvNet/Player.cs#L787-L788), [src/MpvNet/Player.cs#L787-L788](https://github.com/mpvnet-player/mpv.net/blob/ad94042a2c20a4daffaba0574b0c4eebc636c0d4/src/MpvNet/Player.cs#L787-L788), [src/MpvNet/Player.cs#L807-L816](https://github.com/mpvnet-player/mpv.net/blob/ad94042a2c20a4daffaba0574b0c4eebc636c0d4/src/MpvNet/Player.cs#L807-L816).
The handling of `dvb_tele` and `arib` formats comes from my personal testing.

File: lua/dyn_menu.lua [hunk 3]
```diff
@@ -16,9 +16,23 @@
 local menu_prop = 'user-data/menu/items'
 local menu_items = mp.get_property_native(menu_prop, {})
 
+local function esc_for_codec(str)
```
Comment (tsl0922) [inline lua/dyn_menu.lua:19]: Do you think we should lowercase `str` before compare? Not sure if mpv will return uppercased codec.

Comment (dyphire) [reply lua/dyn_menu.lua:19]: No need to do this, mpv provides codec names through ffmpeg, which are all lowercase. You can provide this command of `ffmpeg -codecs` to view the list of codecs supported by the compiled ffmpeg.

**Comment by butterw** [comment]
The display length of the track menu is always going to be a concern #10, but it is also important to display as much useful info as possible.

Replacing default with  * gains a few characters and would match the mpv terminal output, I think.

**Comment by dyphire** [comment]
For the display length of tracks and chapters lists, the main issue is that the title may be too long. We can simply truncate the title length to avoid possible length overflow. I'm just not sure about the choice of string length limit, it should be between 80-150.
Just like this:
```lua
if track.title and track.title:len() > 100 + 5 then
    track.title = track.title:sub(1, 100) .. " ..."
end
......
if title and title:len() > 100 + 5 then
    title = title:sub(1, 100) .. " ..."
end
```

**Comment by butterw** [comment]
These length truncation limits should be made configurable via  --script-opts so at to allow for a more compact menu display if desired.

Often there is no track title, but default is always there.

**Comment by tsl0922** [comment]
Well,  I don't care about it very much, because I' using a 4K monitor.

If you really want to limit it, you may open another PR. Make the limit configurable is a good idea.

**Comment by tsl0922** [comment]
> Replacing default with *

I have changed default to ` (*)` at the end.

**Pull request merged**